boughtbymany/mutt-forms-vue
# feat: Array Slot Removal

# Description

There is currently only an option to display array controls which allow the pushing and popping of slots from the Array field. This PR adds the ability to enable a control to remove a specific slot from an Array field.

The PR also adds callback events when a slot is added/removed which returns the slot added/removed should it need to be used by the event handler.

## Type of change


# How Has This Been Tested?

1. Run the demo with `cd demo && npm run demo`
1. On the demo page there is an Array field rendered.
1. For each slot in the Array field, there should be a **Remove** button
1. Click the **Remove** button for the slot you want to remove.
1. The correct slot should be removed.
1. Test with adding multiple slots and removing random slots to ensure the correct slot is removed.
1. For each slot add/remove, the event will be logged to the console.

# Checklist:

- [x] Designer has reviewed the changes
